### ISSUE TITLE
Option to enable Component gateway in Dataproc

### ISSUE THREAD

**Issue by ChethanUK** [GoogleCloudPlatform/dataproc-initialization-actions#500]
How can I enable Component gateway in dataproc?

Neither the command line option nor the GCP provides a working option:
![Right now there is no option in GCP](https://i.imgur.com/epOZqWX.png)

I tried: `    --enable-component-gateway \ ` but got error:
```
ERROR: (gcloud.dataproc.clusters.create) unrecognized arguments: --enable-component-gateway (did you mean '--optional-components'?) 
```

Any working solution?

**Comment by medb** [comment]
This is a Beta feature, that's why you [should use](https://cloud.google.com/dataproc/docs/concepts/accessing/dataproc-gateways?hl=bg#creating_a_cluster_with_component_gateway) `beta` track in `gcloud` command to use it:
```bash
gcloud beta dataproc clusters create $CLUSTER_NAME \
    --enable-component-gateway \
    . . .
```

**Comment by ChethanUK** [comment]
Why was the case closed? 

I did try out hat but it didn't work

```
ERROR: (gcloud.beta.dataproc.clusters.create) unrecognized arguments```

There are a lot of problems in Dataproc..

**Comment by ChethanUK** [comment]
I even tried various option to create clusters, most permutations possible:

```
gcloud beta dataproc clusters create ${CLUSTERNAME} \
--optional-components=JUPYTER, PRESTO \
--subnet default --zone europe-west2-c --master-machine-type n1-standard-4 --master-boot-disk-size 1000 \
--num-workers 2 --worker-machine-type n1-standard-3 --worker-boot-disk-size 1000 \
--num-preemptible-workers 3 --image-version 1.4-debian9 \
--properties spark:spark.jars.packages=datastax:spark-cassandra-connector:2.4.1-s_2.11 \
--enable-component-gateway ```

I got `ERROR: (gcloud.beta.dataproc.clusters.create) unrecognized arguments: PRESTO`

**Comment by medb** [comment]
Case was closed, because I tested provided command and it worked:
```bash
$ gcloud beta dataproc clusters create <cluster-name> --enable-component-gateway
Waiting on operation [projects/<project>/regions/global/operations/<operation_id>].
Waiting for cluster creation operation...⠹                                                                                                                   
WARNING: For PD-Standard, we strongly recommend provisioning 1TB or larger to ensure consistently high I/O performance. See https://cloud.google.com/compute/docs/disks/performance for information on disk I/O performance.
Waiting for cluster creation operation...done.                                                                                                               
Created [https://dataproc.googleapis.com/v1beta2/projects/<project>/regions/global/clusters/<cluster-name>] Cluster placed in zone [<zone>].
```

**Comment by ChethanUK** [comment]
TRY:
```
gcloud beta dataproc clusters create ${CLUSTERNAME} \
--optional-components=JUPYTER, ANACONDA, PRESTO \
--subnet default --zone europe-west2-c --master-machine-type n1-standard-4 --master-boot-disk-size 1000 \
--num-workers 2 --worker-machine-type n1-standard-3 --worker-boot-disk-size 1000 \
--num-preemptible-workers 3 --image-version 1.4-debian9 \
--properties spark:spark.jars.packages=datastax:spark-cassandra-connector:2.4.1-s_2.11 \
--enable-component-gateway
```

I am getting: 
```
ERROR: (gcloud.beta.dataproc.clusters.create) unrecognized arguments:
  ANACONDA,
  PRESTO
```

**Comment by ChethanUK** [comment]
> because I tested provided command and it worked:

Ya, but Obviously we won't use just enable-component-gateway, We use it along with some optional component, at least JUPYTER

**Comment by medb** [comment]
You have multiple typos in your [command](https://github.com/GoogleCloudPlatform/dataproc-initialization-actions/issues/500#issuecomment-483033514):
* extra whitespace before `PRESTO` component.
* missing `ANACONDA` component that required for `JUPYTER`
* non-existent `n1-standard-3` machine type
* not existent `default` subnetwork in my case

Here is working command:
```bash
gcloud beta dataproc clusters create ${CLUSTERNAME} \
--optional-components=ANACONDA,JUPYTER,PRESTO \
--subnet default --zone europe-west2-c \
--master-machine-type n1-standard-4 --master-boot-disk-size 1000 \
--num-workers 2 --worker-machine-type n1-standard-4 --worker-boot-disk-size 1000 \
--num-preemptible-workers 3 --image-version 1.4-debian9 \
--properties spark:spark.jars.packages=datastax:spark-cassandra-connector:2.4.1-s_2.11 \
--enable-component-gateway
```

While there could be occasional bugs, all your last reports related yo typos and misuses of `gcloud` commands.

**Comment by ChethanUK** [comment]
> * extra whitespace before `PRESTO` component

Woah, Didn't know it will give an error, Atleast error message should be proper then Or maybe some hint like remove the extra whitespace

**Comment by ChethanUK** [comment]
> Please, read Dataproc documentation more carefully!

Ya, but it's not perfect

For example, For example in [Presto Docs](https://cloud.google.com/dataproc/docs/concepts/components/presto), it's not clear how we can expose the server to other components like Apache Superset and a lot of other things are unclear.

**Comment by medb** [comment]
Please, leave feedback about this and any other issue that you noticed in Dataproc documentation (using `SEND FEEDBACK` link on specific documentation page) and we will improve Dataproc documentation to address it.

**Comment by ChethanUK** [comment]
okay!. It will be really nice to have more documentation on PRESTO's 

You know right, we can't give everyone the SSH access to the master node just to run simple presto SQL query or Integrating with Apache Superset etc..

Just installing components is not everything... More examples, tutorials on common use cases and in-depth documentation can help al..

**Comment by ChethanUK** [comment]
l

**Comment by medb** [comment]
Currently we are working on support of Presto job type in Dataproc API, when it released you will be able to submit Presto queries without SSHing into master node, it will be available in coming weeks.

**Comment by jean-lucas** [comment]
@medb 

> Currently we are working on support of Presto job type in Dataproc Jobs API. When it released, you will be able to submit Presto queries without SSHing into master node, it will be available in coming weeks.

Is there any where I can track the progression of this?
